apps/comps/hooks/useUnlockKeys.ts

```diff
@@ -18,45 +19,39 @@ export const useUnlockKeys = (agentHandle: string, agentId?: string) => {
     enabled: !!agentId,
   });
 
-  // Query for sandbox agent - fetch when agent exists
-  const sandboxAgentQuery = useQuery({
-    queryKey: ["sandbox-agent", agentHandle],
-    queryFn: async () => {
-      return await sandboxClient.getAgentApiKey(agentHandle);
-    },
-    enabled: !!agentHandle,
-  });
-
-  // Get sandbox agent id
-  const sandboxAgentId = sandboxAgentQuery.data?.agent?.id;
-
-  // Query for sandbox competitions - fetch when email verified and agent exists
-  const sandboxCompetitionsQuery = useQuery({
-    queryKey: ["sandbox-agent-competitions", sandboxAgentId],
-    queryFn: async () => {
-      if (!sandboxAgentId) throw new Error("Sandbox agent ID required");
-      return await sandboxClient.getAgentCompetitions(sandboxAgentId, {
-        status: "active",
-      });
-    },
-    enabled: !!sandboxAgentId,
-    staleTime: 0, // Always consider data stale
-    refetchOnMount: "always", // Always refetch when component mounts
-  });
-
-  // Query for sandbox API key - fetch when email verified and agent exists in sandbox
+  // Query for sandbox agent API key
   const sandboxKeyQuery = useQuery({
     queryKey: ["sandbox-agent-api-key", agentHandle],
     queryFn: async () => {
       try {
+        // Note: the sandbox-specific API client uses the `handle` since it does additional lookups
+        // to get the agent ID relative to the sandbox (the sandbox vs production agent IDs differ)
         return await sandboxClient.getAgentApiKey(agentHandle);
       } catch (error) {
-        // If agent doesn't exist in sandbox yet, return null instead of failing
+        // If sandbox agent doesn't exist, create it. Note: this is a proactive action in case the
+        // "sync" between the main API and sandbox somehow gets out of sync
         if (
           error instanceof Error &&
           error.message.includes("Agent not found")
         ) {
-          return null;
+          const createResult = await sandboxClient.createAgent({
```

Comment:
if the sandbox agent isnt found, we just create the agent. the UI coordinates sandbox agent creation, and sometimes, it seems it doesnt work as expected. 

this'll help cover that edge case and always get an existing sandbox agent, or create a new one if it doesnt exist.